### ISSUE TITLE
Allow localhost as MySQL host for SSH connections

### ISSUE THREAD

**Issue by kingkool68** [Sequel-Ace/Sequel-Ace#109]
Bringing this over from https://github.com/sequelpro/sequelpro/issues/1146

I need to connect to a MySQL database over SSH that has `skip-name-resolve = 1` set in the configuration. With host set to `127.0.0.1` I can't authenticate and connect. I SSH'd into my server and tried connecting to the database locally like so:

```
ubuntu@root:~$ mysql -u <username> --password <database-name> -h 127.0.0.1
Enter password:
ERROR 1045 (28000): Access denied for user '<username>'@'127.0.0.1' (using password: YES)
```

But when I do this I am able to connect:

```
ubuntu@root:~$ mysql -u <username> --password <database-name> -h localhost
MariaDB [<username>]> 
```

In Sequel Pro I tried changing the host to `localhost` but Sequel Pro always wanted to change it to `127.0.01` and there was nothing I could do about it. 

**It would be nice if I could enter `localhost` as the host value for the database**

See https://community.wordops.net/d/238-allowing-connection-to-database-via-sequel-pro-over-ssh

**Comment by Jason-Morcos** [comment]
Are you unable to connect if you switch the connection type to "socket"? Reading through some of the past commentary on this issue, it doesn't seem very straightforward or clear how to properly handle it.

**Comment by gboudreau** [comment]
Connection using `localhost` on remote is OK. I don't think this check should occur when SSH has been selected.
I tried to connect using the `localhost` hostname, and SSH tunneling, and it worked just fine.
I think the libmysqlclient library has been updated at some point, to "fix" this (maybe it tries using a socket first, and will fallback to using TCP when that fails..?)

**Comment by kingkool68** [comment]
If I try to connect via socket the SSH options disappear. When SSH is selected if I enter `localhost` for the host value this popup happens as soon as I click away to a different field.

![image](https://user-images.githubusercontent.com/867430/85814979-452dca80-b735-11ea-8b5f-5e1ed4c194a9.png)

**Comment by gboudreau** [comment]
@kingkool68 Is your MySQL server only listening for incoming connections on a socket, and not using a TCP port?
If that is the case, allowing the `localhost` hostname won't help you, because Sequel Ace always connects using TCP, when using SSH tunneling. It has no option to tunnel a socket connection.

**Comment by kingkool68** [comment]
I'm not sure if my MySQL server is only listening for incoming connections on a socket. I just know that if I remove the `skip-name-resolve = 1` from the MySQL conf I am able to connect to the database over SSH using Sequel Pro/Sequel Ace.

I use another open source project to aide in managing my server administration and the project says you should be able to use `localhost` as the MySQL host. I just thought I would raise this issue as it is a minor annoyance to me.

**Comment by KittMedia** [comment]
By using `skip-name-resolve = 1` the MySQL server is only binding on `localhost`, not on `127.0.0.1` even if it’s usable via TCP port.

**Comment by KittMedia** [comment]
By using `skip-name-resolve = 1` the MySQL server is only binding on `localhost`, not on `127.0.0.1` even if it’s usable via TCP port.

**Comment by gboudreau** [comment]
> By using `skip-name-resolve = 1` the MySQL server is only binding on `localhost`, not on `127.0.0.1` even if it’s usable via TCP port.

Could you link to the documentation that states this? I did a quick search a few days ago, and couldn't find any relation between `skip-name-resolve` and whether MySQL would listen on TCP port or not.
Thx.

**Comment by KittMedia** [comment]
Here: http://download.nust.na/pub6/mysql/doc/refman/5.1/en/dns.html

> You can disable DNS host name lookups by starting mysqld with the --skip-name-resolve option. However, in this case, you can use only IP numbers in the MySQL grant tables.

The `skip-name-resolve` has nothing to do with the ability of MySQL to connect via TCP or socket, it just disables DNS lookups. And if the IP address `127.0.0.1` is not within the MySQL grant tables, it just can connect via `localhost`.

**Comment by gboudreau** [comment]
Ah, ok. Then the fix for this is trivial. I'll send a PR in a few minutes.

**Comment by Jason-Morcos** [comment]
@kingkool68 Could you try this latest beta and let us know if it works? @gboudreau made that change so you should be able to connect to "localhost". Keep us posted!

https://github.com/Sequel-Ace/Sequel-Ace/releases/tag/2.1.1-beta1

**Comment by kingkool68** [comment]
I tried the beta but I'm having general connection issues mentioned in https://github.com/Sequel-Ace/Sequel-Ace/issues/141#issuecomment-652770866

**Comment by Jason-Morcos** [comment]
@kingkool68 Unfortunately it looks like the attempted fix to MySQL auth for issue #141 broke some other connections. Reverted. How about trying this beta? https://github.com/Sequel-Ace/Sequel-Ace/releases/tag/2.1.1-beta2

**Comment by kingkool68** [comment]
@Jason-Morcos beta 2 fixed my connection issues.
